augment side-channel topic-stub to full-fledged topic (#1019)
bitcoinops/bitcoinops.github.io: _topics/en/side-channels.md

# see_also:
#   - title:
#     link:
---

> Side channel attacks can be generally categorized as timing attacks, power-monitoring attacks, acoustic cryptanalysis, electromagnetic attacks, and cache attacks. Statistical methods and machine learning algorithms are often used in combination with physical access to a hardware signer in order to carry out attacks. E.g. an attacker can take a factory hardware signer, gather a dataset that measure changes in power consumption during critical calculations on this signer, then use this dataset on a new signer, measure this new signer's power consumption during it's critical calculations, and exfiltrate the passcode by comparing the power consumption of the new signer with the statistical analysis done on the factory signer.

Reviewer: Review would be easier if we limit line lengths.
```suggestion
Side channel attacks can be generally categorized as
- timing attacks
- power-monitoring attacks
- acoustic cryptanalysis
- electromagnetic attacks
- cache attacks

Statistical methods and machine learning algorithms are often used in combination with physical access to a hardware signer in order to carry out attacks. For example, an attacker can take a factory hardware signer, gather a dataset that measures changes in power consumption during critical calculations on this signer, then use this dataset on a new signer, measure this new signer's power consumption during its critical calculations, and exfiltrate the passcode by comparing the power consumption of the new signer with the statistical analysis done on the factory signer.
```

> Some side channel attack surfaces in the Bitcoin ecosystem have been mitigated or precluded entirely. E.g. the libsecp256k1 library has merged the safegcd algorithm into its codebase. The safegcd algorithm is a way to calculate modular inverse operations in constant time. A modular inverse operation is a logical operation used when verifying ECDSA signatures.

Reviewer 2: ```suggestion
Some side channel attack surfaces in the Bitcoin ecosystem have been mitigated or precluded entirely. E.g. the libsecp256k1 library has merged the SafeGCD algorithm into its codebase. The SafeGCD algorithm is a way to calculate modular inverse operations in constant time. A modular inverse operation is a logical operation used when verifying ECDSA signatures.
```

> Some side channel attack surfaces in the Bitcoin ecosystem have been mitigated or precluded entirely. E.g. the libsecp256k1 library has merged the SafeGCD algorithm into its codebase. The SafeGCD algorithm is a way to calculate modular inverse operations in constant time. A modular inverse operation is a logical operation used when verifying ECDSA signatures.

Reviewer: ```suggestion
Some side channel attack surfaces in the Bitcoin ecosystem have been mitigated or precluded entirely. For example, the libsecp256k1 library has merged the SafeGCD algorithm into its codebase. The SafeGCD algorithm is a way to calculate modular inverse operations in constant time. A modular inverse operation is a logical operation used when verifying ECDSA signatures.
```

> TROPIC01 is an auditable RISCV secure chip under development, with intention to release as part of a hardware signer. The usage of secure chips may mitigate some side channel attacks that hardware signers are currently vulnerable towards.

Reviewer: ```suggestion
TROPIC01 is an auditable RISCV secure chip under development, with the intention to release it as part of a hardware signer. Using secure chips may mitigate some side channel attacks that hardware signers are currently vulnerable to.
```

> Tropic Square, a SatoshiLabs/Trezor startup, has built and tested the TROPIC01, an auditable RISCV secure chip, with intention to release as part of a hardware signer. The usage of secure chips may mitigate some side channel attacks that hardware signers are currently vulnerable towards.

Author: Your suggestion is fair. I was wondering if my original version sounded too much like an advert for Trezor.

# see_also:
#   - title:
#     link:
---
Side channel attacks can be generally categorized as
- timing attacks
- power-monitoring attacks
- acoustic cryptanalysis
- electromagnetic attacks
- cache attacks

> Statistical methods and machine learning algorithms are often used in combination with physical access to a hardware signer in order to carry out attacks. For example, an attacker can take a factory hardware signer, gather a dataset that measures changes in power consumption during critical calculations on this signer, then use this dataset on a new signer, measure this new signer's power consumption during its critical calculations, and exfiltrate the passcode by comparing the power consumption of the new signer with the statistical analysis done on the factory signer.

Reviewer 3: I’m not sure what exactly to make of “factory hardware signer”, would “factory-fresh” perhaps express what you mean? Also, I felt that “new” was an odd adjective to refer to the target of the attack. What do you think about:

```suggestion
Statistical methods and machine learning algorithms are often used in combination with physical access to a hardware signer in order to carry out attacks. For example, an attacker can take a factory-fresh hardware signer to measure changes in power consumption during critical calculations, then use this dataset to exfiltrate the key material from a second signer by comparing its power consumption with the statistical analysis done on the factory-fresh signer.
```

Author: I think this is much better. thanks.
The only thing is that a factory-fresh signer doesn't actually have to be factory fresh, it could be any signer laying about, so maybe the best word choice is 'a __sample__ hardware signer' or 'an __available__ hardware signer'

Reviewer 3: Yeah, that makes more sense to me :+1:

Some side channel attack surfaces in the Bitcoin ecosystem have been mitigated or precluded entirely. For example, the libsecp256k1 library has merged the SafeGCD algorithm into its codebase. The SafeGCD algorithm is a way to calculate modular inverse operations in constant time. A modular inverse operation is a logical operation used when verifying ECDSA signatures.

> TROPIC01 is an auditable RISCV secure chip under development, with the intention to release it as part of a hardware signer. Using secure chips may mitigate some side channel attacks that hardware signers are currently vulnerable to.

Reviewer 4: I think this first sentence is too specific about something we haven't discussed in the newsletter.  I think it's ok to drop it and just use the second sentence here.

Reviewer 2: @paulkania if you agree with this suggestion, would you mind making the change?

Reviewer 2: @paulkania going to close this for now until we resolve this suggestion. feel free to re-open this PR!
